src/ProjectService.Data/ProjectRepository.cs

```diff
@@ -14,6 +15,28 @@ public class ProjectRepository : IProjectRepository
     {
         private readonly IDataProvider provider;
 
+        private IQueryable<DbProject> CreateFindPredicates(
+            FindDbProjectsFilter filter,
+            IQueryable<DbProject> dbProjects)
+        {
+            if (!string.IsNullOrEmpty(filter.Name))
+            {
+                dbProjects = dbProjects.Where(u => u.Name.ToUpper().Contains(filter.Name.ToUpper()));
+            }
+
+            if (!string.IsNullOrEmpty(filter.ShortName))
+            {
+                dbProjects = dbProjects.Where(u => u.ShortName.ToUpper().Contains(filter.ShortName.ToUpper()));
+            }
+
+            if (filter.IdNameDepartments != null)
+            {
+                dbProjects = dbProjects.Where(u => filter.IdNameDepartments.Keys.Contains(u.DepartmentId));
+            }
```

Comment:
Why not used LinqKit?

Reply:
What do you want to see here?

Reply:
I'm not sure if the name and department will always be unique for the project. Why not search for both?